StreamPracticeSet/src/main/java/Metrics.java

```diff
@@ -0,0 +1,72 @@
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Metrics {
+    private final List<Order> orders;
+
+    public Metrics(List<Order> orders) {
+        this.orders = Objects.requireNonNullElse(orders, Collections.emptyList());
+    }
+
+    public List<String> getUniqueCities() {
+        Set<String> uniqueCities = orders.stream()
+                .map(Order::getCustomer)
+                .filter(Objects::nonNull)
+                .map(Customer::getCity)
+                .filter(Objects::nonNull)
```

Comment:
There are too many explicit filter(Objects::nonNull) calls. A cleaner approach would be to ensure data integrity at the source (when the Order objects are created) rather than scattering null checks throughout the stream pipelines.